wd_helper.au3

```diff
@@ -672,6 +679,177 @@ Func _WD_FrameLeave($sSession)
+
+			; recalculate locations from absolute path on COL0 to relative path on COL1
+			$a_Result[$i][$_WD_FRAMELIST_Relative] = StringRegExpReplace($a_Result[$i][$_WD_FRAMELIST_Absolute], '\A' & $sStartLocation & '\/?', '')
+		Next
+
+		; select desired DataType for the $vResult - usually string is option for testing and asking support
+		If $bReturnAsArray Then
+			$vResult = $a_Result
+		Else
+			$vResult = _ArrayToString($a_Result)
+			If @error Then
+				$iErr = $_WD_ERROR_RetValue
+				$sMessage &= ' ArrayToSring conversion failed'
+				$vResult = ''
+			EndIf
+		EndIf
+
+	EndIf
+
+	; Back to "calling frame"
+	If $sStartLocation Then
+		_WD_FrameEnter($sSession, $sStartLocation)
+		$iErr = @error
+	EndIf
+
+	If $sStartLocation = '' Or ($iErr And $iErr <> $_WD_ERROR_RetValue) Then
+		$sMessage &= ' Was not able to check / back to "calling frame"'
+	EndIf
+
+	#EndRegion - post processing
+
+	$_WD_DEBUG = $_WD_DEBUG_Saved ; restore DEBUG level
+
+	$sMessage = ($sMessage And $_WD_DEBUG > $_WD_DEBUG_Error) ? (' Information: ' & $sMessage) : ("")
+	Return SetError(__WD_Error($sFuncName, $iErr, $sParameters & $sMessage), 0, $vResult)
+EndFunc   ;==>_WD_FrameList
+
+; #INTERNAL_USE_ONLY# ===========================================================================================================
+; Name ..........: __WD_FrameList_Internal
+; Description ...: function that is used internally in _WD_FrameList, even recursively when nested frames are available
+; Syntax ........: __WD_FrameList_Internal($sSession, $sLevel, $sFrameAttributes, $_WD_DEBUG_Saved)
+; Parameters ....: $sSession            - Session ID from _WD_CreateSession
+;                  $sLevel              - frame location level ... path
+;                  $sFrameAttributes    - all <iframe ....> atributes
+;                  $_WD_DEBUG_Saved     - log level taken from calling function
+; Return values .: Success - array or string
+;                  Failure - "" (empty string) and sets @error to one of the following values:
+;                  - $_WD_ERROR_Exception
+; Author ........: mLipok
+; Modified ......: Danp2
+; Remarks .......:
+; Related .......: _WD_FrameList, _WD_GetFrameCount, _WD_FrameEnter, _WD_FrameLeave
+; Link ..........:
+; Example .......: No
+; ===============================================================================================================================
+Func __WD_FrameList_Internal($sSession, $sLevel, $sFrameAttributes, $_WD_DEBUG_Saved)
+	Local Const $sFuncName = "__WD_FrameList_Internal"
+	Local Const $sParameters = 'Parameters:    Level=' & $sLevel ; intentionally $sFrameAttributes is not listed here to not put too many data into the log
+	Local $iErr = $_WD_ERROR_Success
+	Local $vResult = '', $s_URL = '', $sMessage = ''
+
+	_WD_FrameEnter($sSession, $sLevel)
+	If $sLevel = 'null/3' Then SetError($_WD_ERROR_Exception)
```

Comment:
this is only for testing should be removed before merging

Reply:
Take a look on: [intentional testing error removed](https://github.com/Danp2/au3WebDriver/pull/362/commits/14ff948ec2734e80027648b2136c6e53fa54b25e)

Treat this part of conversationa as resolved.